### PR TITLE
add checking logic for getPortalClientRect and onMentionSelect.

### PR DESCRIPTION
**Pull request by mzbac**
fix issues from https://github.com/draft-js-plugins/draft-js-plugins/issues/388

### CODE REVIEW

File: draft-js-mention-plugin/src/MentionSuggestions/index.js [hunk 1]
```diff
@@ -206,6 +208,9 @@ export default class MentionSuggestions extends Component {
   };
 
   onMentionSelect = (mention) => {
+    if (!mention) {
```
Comment (nikgraf) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: In which situation can mention be `null` or `undefined`?

File: draft-js-mention-plugin/src/MentionSuggestions/index.js [hunk 2]
```diff
@@ -58,7 +58,9 @@ export default class MentionSuggestions extends Component {
           focusedOptionIndex: size - 1,
         });
       }
-
+      if (!this.props.store.getAllSearches().has(this.activeOffsetKey)) {
```
Comment (nikgraf) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: I'm bit out of the this code. Can you elaborate why this is solving the problem? At best add a comment so we know in the future when revisit it.

Comment (mzbac) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: it doesn't solve the multi-character mention trigger problem.  it just a simple protection for the error when componentDidUpdate try to get new getPortalClientRect but key already deleted by previous action. ( right now, it only can happened when set the mention trigger to be multi-characters. current mention plugin logic is kind of handle mention trigger as single char. for example:
`onSearchChange = (editorState, selection) => {
    const { word } = getSearchText(editorState, selection);
    const searchValue = word.substring(1, word.length);
    if (this.lastSearchValue !== searchValue) {
      this.lastSearchValue = searchValue;
      this.props.onSearchChange({ value: searchValue });
    }
  };`
`const selectionIsInsideWord = leaves
      .filter((leave) => leave !== undefined)
      .map(({ start, end }) => (
        start === 0 && anchorOffset === 1 && anchorOffset <= end || // @ is the first character
        anchorOffset > start + 1 && anchorOffset <= end // @ is in the text or at the end
      ))`

no sure if we need add support for mention trigger of multiple character

File: draft-js-mention-plugin/src/MentionSuggestions/index.js (hunk 1 above)
Comment (mzbac) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: user didn't select mention from mentionsuggestions, instead they can just by typing @xxx (invalidate mentions) then hit enter .. in that case mention will be undefined..

Comment (nikgraf) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: ohhh, good point

File: draft-js-mention-plugin/src/MentionSuggestions/index.js (hunk 2 above)
Comment (nikgraf) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: ok, so then the plugin doesn't throw an error, but won't work anyway. is this correct?

Comment (mzbac) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: yes, it won't work for mention trigger more than 1 char. but at least the code doesn't break now..I think we may either add docs to tell user that mention trigger only can be set as single character or     add support for mulit-character mention trigger.

**Comment by nikgraf** [comment]
@mzbac I updated the docs already in another PR

**Comment by evankline** [comment]
@nikgraf @mzbac I believe a similar error is occurring when the MentionSuggestions are unfocused via a click or escape and the trigger character is still in the input. From what I observed it seems to be that the MentionSuggestionsPortal's searchPortal ref is no longer populated. I also noticed that if the same steps are taken, the same error occurs if a ref has been added to the Editor component as in the example below. On what may be a separate issue, or possibly a related issue, when the input is re-focused with the trigger character to the left of the cursor, would it be expected behavior to have the MentionSuggestions component opened as though the user had just typed the trigger character?

```
<Editor 
  editorState={this.state.editorState} 
  ref={(el) => this.editor = el} 
  plugins={plugins}
  onChange={this.onChange}
/>

File: draft-js-mention-plugin/src/MentionSuggestions/index.js (hunk 2 above)
Comment (nikgraf) [inline draft-js-mention-plugin/src/MentionSuggestions/index.js]: I did that :)

**Pull request merged**